File: src/main/java/com/example/librarymanagement/service/JournalService.java
package com.example.librarymanagement.service;

import com.example.librarymanagement.model.entity.Journal;

import java.util.List;

public interface JournalService {
    Journal createJournal(Journal journal);
    Journal getJournalById(Long id);
    List<Journal> getJournalByBookPresenceId(Long bookPresenceId);
    Journal getJournalByBookPresenceIdAndUserId(Long bookPresenceId, Long userId);
    List<Journal> getJournalByUserId(Long userId);
    void deleteJournal(Long id);
}

File: src/main/java/com/example/librarymanagement/service/JournalService.kt
package com.example.librarymanagement.service

import com.example.librarymanagement.model.domain.Journal
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface JournalService {
    fun save(journal: Journal): Mono<Journal>
    fun getJournalById(id: String): Mono<Journal>
    fun findByBookPresenceIdAndUserIdAndDateOfReturningIsNull(bookPresenceId: String, userId: String): Mono<Journal>
    fun getJournalByUserId(userId: String): Flux<Journal>
    fun deleteJournalById(journalId: String): Mono<Unit>
}
